var/spack/repos/builtin/packages/hpctoolkit-externals/package.py

```diff
@@ -0,0 +1,45 @@
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class HpctoolkitExternals(Package):
+
+    """
+    HPCToolkit performance analysis tool has many prerequisites and
+    HpctoolkitExternals package provides all these prerequisites.
+    """
```

Comment:
Yeah, I still need to document that. Waiting on #1676 and #1664 to be merged before I muck around with that too much.

To run the flake8 tests, you need to install flake8. This can be done by running `spack install py-flake8`. You need flake8 in your PATH, and py-setuptools needs to be in your PYTHONPATH. Once everything is installed correctly, you can run `share/spack/qa/run-flake8-tests` to test everything.

Running the documentation tests is a similar process. It requires py-sphinx, graphviz, git, subversion, and mercurial to be installed. Sphinx requires `py-setuptools` to be in its PYTHONPATH. Once you have all the necessary dependencies installed and in your PATH, simply run `share/spack/qa/run-doc-tests`.

Let me know if you have any questions or run into any problems.